manuskript/ui/editors/editorWidget.py

```diff
@@ -344,12 +352,33 @@ def modelDataChanged(self, topLeft, bottomRight):
         if topLeft.row() <= self.currentIndex.row() <= bottomRight.row():
             self.updateStatusBar()
 
-    #def rowsAboutToBeRemoved(self, parent, first, last):
-        #if self.currentIndex:
-            #if self.currentIndex.parent() == parent and \
-                                    #first <= self.currentIndex.row() <= last:
-                ## Item deleted, close tab
-                #self.mw.mainEditor.tab.removeTab(self.mw.mainEditor.tab.indexOf(self))
+    def rowsInserted(self, parent, first, last):
+        if self.currentIndex:
+            self.updateIndexFromID(fallback=self.currentIndex.parent())
+
+    def rowsAboutToBeRemoved(self, parent, first, last):
+        if not self.currentIndex:
+            return
+
+        # Check parents until we find a parent that matches the parent we are removing.
+        crumb = self.currentIndex
+        commonality = crumb.parent() # start at folder above current text
+        while (commonality != parent):
+            crumb = commonality
+            commonality = commonality.parent()
+
+            if commonality == None:
+                return  # we ran out of parents without finding the matching QModelIndex
+
+            if commonality == parent:
+                break  # we found our shared parent
```

Comment:
Did you confirm if `self._model.rootItem` == a non valid QModelIndex ? Test with moving/deleting scenes that are in the root directly.

Reply:
I tried the code with items in the highest root level and I didn't see any problems while cutting or moving them. That said, I have not specifically found a way to see if rootItem can pop up in that situation.

rootItem is defined as a special Outline item with ID 0, by the way. See abstractModel.py. I do not believe we should be able to come across it as a parent because the only way for it to end up that way is if the rest of the code somehow set the root item as a parent... but all the model code checks for rootItem, so that should not happen.